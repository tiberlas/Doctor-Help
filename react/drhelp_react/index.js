const express = require('express');
const app = express();

require('./routes')(app); //these are your api routes

//Non api requests in production
if (process.env.NODE_ENV === 'production' || process.env.NODE_ENV === 'staging') {
    // Add production middleware such as redirecting to https

    // Express will serve up production assets i.e. main.js
    app.use(express.static(__dirname + '/client/build'));
    // If Express doesn't recognize route serve index.html
    const path = require('path');
    app.get('*', (req, res) => {
        res.sendFile(
            path.resolve(__dirname, 'client', 'build', 'index.html')
        );
    });
}

//start server
const PORT = process.env.PORT || 5000; //Heroku sets port dynamically
app.listen(PORT, () => {
    console.log('listening...');
}).on('error', err => {
    console.log(`Error Code: ${err.code}`);
});