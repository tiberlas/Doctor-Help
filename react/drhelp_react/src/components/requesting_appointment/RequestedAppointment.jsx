import React, { Component } from 'react';
import axios from 'axios';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import RoomList from './RoomList';

class RequestedAppointment extends Component {
    state = {  
        id: '',
        date: '',
        procedure: '',
        typeId: 0,
        doctor: '',
        nurse: '',
        patient: ''
    }

    componentDidMount() {
        let path_parts = window.location.pathname.split('appointment/')
        axios.get('http://localhost:8080/api/appointments/requests/id=' + path_parts[1])
            .then(responce => {
                this.setState({
                    id: responce.data.id,
                    date: responce.data.date,
                    procedure: responce.data.type,
                    doctor: responce.data.doctor,
                    nurse: responce.data.nurse,
                    patient: responce.data.patient,
                    typeId: responce.data.typeId
                })
            })
    }

    render() {
        return (
            <div>
            <div class='row d-flex justify-content-center'>
            <div class='col-md-7'> 
                <br/>
                <Table class="table table-hover ">
                    <TableHead class="table-active">
                        <TableRow class="table-active">
                            <TableCell colSpan='5' class='text text-success text-center' align='center'>Requested appointment</TableCell>
                        </TableRow>
                        <TableRow class="table-active">
                            <TableCell class="text-success cursor-pointer" >date and time</TableCell>
                            <TableCell class="text-success cursor-pointer" >procedure</TableCell>
                            <TableCell class="text-success cursor-pointer" >doctor</TableCell>
                            <TableCell class="text-success cursor-pointer" >nurse</TableCell>
                            <TableCell class="text-success cursor-pointer" >patient</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        <TableCell class='text text-white'>{this.state.date}</TableCell>
                        <TableCell class='text text-white'>{this.state.procedure}</TableCell>
                        <TableCell class='text text-white'>{this.state.doctor}</TableCell>
                        <TableCell class='text text-white'>{this.state.nurse}</TableCell>
                        <TableCell class='text text-white'>{this.state.patient}</TableCell>
                    </TableBody>
                </Table>
            </div>
            </div>

                <RoomList date={this.state.date} type={this.state.typeId}></RoomList>

            </div>
        );
    }
}
 
export default RequestedAppointment;