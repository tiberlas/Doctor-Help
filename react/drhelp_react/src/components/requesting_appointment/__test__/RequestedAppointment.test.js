import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import RequestedAppointment from "../RequestedAppointment.jsx";
import { ExpansionPanelActions } from "@material-ui/core";

const value = {
	id: 1,
	date: "02/12/2020 14:56",
	procedure: "vadjenje kamenja iz bubrega",
	duration: "01:20",
	doctor: "Pera Peric",
	nurse: "Ana Nasta",
	patient: "Zrtveno Jagnje",
};

afterEach(cleanup);
test("render without crashing", () => {
	const div = document.createElement("div");

	ReactDOM.render(<RequestedAppointment value={value} />, div);
	ReactDOM.unmountComponentAtNode(div);
});

test("matches snapshot", () => {
	const requiredAppointment = renderer
		.create(<RequestedAppointment value={value} />)
		.toJSON();

	expect(requiredAppointment).toMatchSnapshot();
});
