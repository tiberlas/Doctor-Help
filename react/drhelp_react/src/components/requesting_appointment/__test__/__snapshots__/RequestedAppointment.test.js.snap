// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="row d-flex justify-content-center"
  >
    <div
      class="col-md-7"
    >
      <br />
      <table
        class="table table-hover "
        className="MuiTable-root"
      >
        <thead
          class="table-active"
          className="MuiTableHead-root"
        >
          <tr
            class="table-active"
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              class="text text-success text-center"
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              colSpan="5"
              scope="col"
            >
              Requested appointment
            </th>
          </tr>
          <tr
            class="table-active"
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              date and time
            </th>
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              procedure
            </th>
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              doctor
            </th>
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              nurse
            </th>
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              patient
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <td
            aria-sort={null}
            class="text text-white"
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
          <td
            aria-sort={null}
            class="text text-white"
            className="MuiTableCell-root MuiTableCell-body"
          >
            
             
            
             H
          </td>
          <td
            aria-sort={null}
            class="text text-white"
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
          <td
            aria-sort={null}
            class="text text-white"
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
          <td
            aria-sort={null}
            class="text text-white"
            className="MuiTableCell-root MuiTableCell-body"
          >
            
          </td>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="row d-flex justify-content-center"
  >
    <div
      class="col-md-7"
    >
      <h4>
        List of rooms
      </h4>
      <br />
      <table
        class="table table-hover "
        className="MuiTable-root"
      >
        <thead
          class="table-active"
          className="MuiTableHead-root"
        >
          <tr
            class="table-active"
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              onClick={[Function]}
              scope="col"
            >
              room name
              
            </th>
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              onClick={[Function]}
              scope="col"
            >
              number
              
            </th>
            <th
              aria-sort={null}
              class="text-success cursor-pointer"
              className="MuiTableCell-root MuiTableCell-head"
              onClick={[Function]}
              scope="col"
            >
              first free date
              
            </th>
            <th
              aria-sort={null}
              class="text-success"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <button
                class="btn btn-success"
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                show all
              </button>
            </th>
            <th
              aria-sort={null}
              class="text-success"
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <button
                class="btn btn-success"
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                search
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</div>
`;
